File: src/main/java/com/jhl/mds/consts/MySQLInsertMode.java
package com.jhl.mds.consts;

import com.jhl.mds.jsclientgenerator.JsClientDTO;
import lombok.Getter;

@JsClientDTO(fileName = "mysql-insert-mode", className = "MySQLInsertMode")
public enum MySQLInsertMode {

    INSERT("INSERT"),
    INSERT_IGNORE("INSERT IGNORE"),
    REPLACE("REPLACE");

    @Getter
    private String syntax;

    MySQLInsertMode(String syntax) {
        this.syntax = syntax;
    }
}

File: src/main/resources/static/resources/js/source/dto/my-sqlinsert-mode.js
// @flow


export class MySQLInsertMode {

    /**
     * @type {MySQLInsertMode}
     */
    static INSERT : ?MySQLInsertMode = null;
    /**
     * @type {MySQLInsertMode}
     */
    static INSERT_IGNORE : ?MySQLInsertMode = null;
    /**
     * @type {MySQLInsertMode}
     */
    static REPLACE : ?MySQLInsertMode = null;
    /**
     * @type {string}
     */
    name : string = '';

    /**
     * @param name {string}
     */
    constructor(name : string) {
        this.name = name;
    }

     toJSON() {
          return this.name;
     }
}

MySQLInsertMode.INSERT = new MySQLInsertMode('INSERT');
MySQLInsertMode.INSERT_IGNORE = new MySQLInsertMode('INSERT_IGNORE');
MySQLInsertMode.REPLACE = new MySQLInsertMode('REPLACE');

